src/orange/construct/CsgTypes.cc

```diff
@@ -0,0 +1,87 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/construct/CsgTypes.cc
+//---------------------------------------------------------------------------//
+#include "CsgTypes.hh"
+
+#include <iostream>
+#include <sstream>
+
+#include "corecel/io/Join.hh"
+
+namespace celeritas
+{
+namespace csg
+{
+//---------------------------------------------------------------------------//
+std::ostream& operator<<(std::ostream& os, True const&)
+{
+    os << "true";
+    return os;
+}
+
+//---------------------------------------------------------------------------//
+std::ostream& operator<<(std::ostream& os, False const&)
+{
+    os << "false";
+    return os;
+}
+
+//---------------------------------------------------------------------------//
+std::ostream& operator<<(std::ostream& os, Aliased const& n)
+{
+    os << "->{" << n.node.unchecked_get() << '}';
```

Comment:
I thought that when reading them, `->{3}` better conveys the meaning of "is a pointer to node 3" rather than "=3" which might mean "is surface 3" or something. I wanted to keep the JSON values a single character. Same with "not" etc. I could further expand this to "pointer to node 3" or something to further improve readability, but I also wanted to keep it compact.